### PR TITLE
[KRYS-5667] Update warning message for eth

### PR DESCRIPTION
**Pull request by MrQKyber**

### CODE REVIEW

File: EarnModule/EarnModule/ViewControllers/StakingViewController.swift [hunk 1]
```diff
@@ -163,6 +166,15 @@ class StakingViewController: InAppBrowsingViewController {
         self.rateInfoView.setValue(value: self.viewModel.displayRate)
     }
     
+    fileprivate func updateUIETHWarningView() {
+        guard viewModel.token.address == "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee" else {
```
[the remark this answers is not in the document]
Comment (tungnguyen20) [reply EarnModule/EarnModule/ViewControllers/StakingViewController.swift:170]: After Shanghai, we will turn off that flag

**Comment by tungnguyen20** [comment]
Missing show message when click Unstake on Portfolio screen

File: Dependencies/Dependencies/Protocols/FeatureFlag.swift [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+//  FeatureFlag.swift
+//  Dependencies
+//
+//  Created by Tạ Minh Quân on 24/11/2022.
+//
+
+import Foundation
+
+public protocol FeatureFlag {
+    func showFeature(forKey flagKey: String) -> Bool
```
Comment (tungnguyen20) [inline Dependencies/Dependencies/Protocols/FeatureFlag.swift:11]: I think it should be `isFeatureEnabled(forKey:)`

File: EarnModule/EarnModule/ViewControllers/StakingViewController.swift [hunk 3]
```diff
@@ -164,6 +167,18 @@ class StakingViewController: InAppBrowsingViewController {
         self.rateInfoView.setValue(value: self.viewModel.displayRate)
     }
     
+    fileprivate func updateUIETHWarningView() {
+        guard AppDependencies.featureFlag.showFeature(forKey: FeatureFlagKeys.unstakeWarning) else {
+            return
+        }
+        guard viewModel.token.address == "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee" else {
```
Comment (tungnguyen20) [inline EarnModule/EarnModule/ViewControllers/StakingViewController.swift:174]: Please move "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee" to a constant

File: Dependencies/Dependencies/Protocols/FeatureFlag.swift (hunk 2 above)
Comment (MrQKyber) [reply Dependencies/Dependencies/Protocols/FeatureFlag.swift:11]: use the same function name in App module, so no need change it

Comment (tungnguyen20) [reply Dependencies/Dependencies/Protocols/FeatureFlag.swift]: I just rebased because there is a FeatureFlag on main branch yet. Please check it out.

**Comment by tungnguyen20** [comment]
Screen Shot 2022-11-28 at 11 40 47
Look like you are missing this requirement

File: EarnModule/EarnModule/Resources/Localizable.strings [hunk 4]
```diff
@@ -87,3 +87,4 @@
 "my.supply" = "My supply";
 "unstaking.in.progress" = "Unstaking in progress";
 "unwrap.token" = "Unwrap %@";
+"unstake.warning.pop.up" = "Currently, unstaking is not available, as this functionality is not yet supported by Ethereum./nUnstaking will be enabled after the Shanghai upgrade that will follow The Merge in approximately 6-12 months (~Q2/Q3 2023)";
```
Comment (tungnguyen20) [inline EarnModule/EarnModule/Resources/Localizable.strings:90]: I think it's \n, not /n

File: EarnModule/EarnModule/Resources/Strings.swift [hunk 5]
```diff
@@ -132,4 +132,5 @@ struct Strings {
     static let supplySummary = "supply.summary".toBeLocalised()
     static let mySupply = "my.supply".toBeLocalised()
     static let unstakingInProgress = "unstaking.in.progress".toBeLocalised()
+    static let unstakeWarningPopupMessage = "unstake.warning.pop.up".toBeLocalised()
```
Comment (tungnguyen20) [inline EarnModule/EarnModule/Resources/Strings.swift:135]: Should be `unstakeEthWarningPopupMessage`

**Review by tungnguyen20 — APPROVED** [review]

**Pull request merged**